simulation/test_torrent_status.cpp

```diff
@@ -41,62 +41,139 @@ using namespace sim;
 namespace lt = libtorrent;
 
 // this is a test for torrent_status time counters are correct
+// it is a long running test, we did already try to improve it:
+// - less slow peers vs many fast peers
+// - slow peers with smal torrent vs fast peers with large torrent
+// - less peers with large torrents vs more peers with smal torrent
+// - connection limit vs parallel requests
+// as conlusion the following combination take the least amount of
+// execution time per tick:
+// - many slow peers with small torrent and connection limit
 TORRENT_TEST(status_timers)
 {
-	lt::time_point start_time;
+	lt::time_point startseed_time;
+	lt::time_point last_active_time;
 	lt::torrent_handle handle;
-
-	setup_swarm(1, swarm_test::upload
+	int last_tick;
+	// at least 11s intervall because after resume it takes 11s untill we unchuck
+	// new pieces at this slow upload rate
+	int ticks_check_intervall = 20;
+	seconds active_time = seconds(0);
+	bool tick_is_in_active_range = false;
+
+	setup_swarm(10, swarm_test::upload_slow
```

Comment:
Because the same torrent need to get multible Upload time updates to check that an update does work before and after shift of timebase.

Network packet amount is the same for this setup comparation, I just need default torrentsize * 9 / 2KiB rate to get enoght upload ticks for the test. Could be one large torrent or 9 times a small torrent.
I was surprised about that fact that a large torrent * 1 takes more execution time then small torrent * 9, too.
It was not mutch more, We have 9% less piece hash checks with the small torrent because seeder has to do 90% less hash checks and we need smaller cache sizes per session with smaller torrents and we use more threads with multible nodes. In sum all these things save more execution time then we have to spend for the overhead of 8 additional nodes.

Let's make the decision large torrent * 1 vs small torrent * 9 based on the torrent_peer timestamp test implementation. If it is to difficould to test 9 peers because I don't know when the do anything, then I go for nodes 2, otherwise I would use nodes 10 to test multiple peer timer changes.